controllers/concept.go

```diff
@@ -225,15 +248,27 @@ func (u ConceptController) RetrieveAttritionTable(c *gin.Context) {
 
 		header := headerAndNonFilteredRow[0]
 		sortedConceptValues := header[2:]
-		filteredRows, err := u.GetFilteredConceptRows(sourceId, cohortId, conceptIds, breakdownConceptId, sortedConceptValues)
+		// append concepts to attrition table:
+		conceptVariablesAttritionRows, err := u.GetConceptVariablesAttritionRows(sourceId, cohortId, conceptIds, breakdownConceptId, sortedConceptValues)
 		if err != nil {
 			log.Printf("Error: %s", err.Error())
 			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error retrieving concept breakdown with filtered conceptIds", "error": err.Error()})
 			c.Abort()
 			return
 		}
+		// append custom dichotomous items to attrition table:
+		customDichotomousVariablesAttritionRows, err := u.GetCustomDichotomousVariablesAttritionRows(sourceId, cohortId, conceptIds, cohortPairs, breakdownConceptId, sortedConceptValues)
+		if err != nil {
+			log.Printf("Error: %s", err.Error())
+			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error retrieving concept breakdown with custom dichotomous variables (aka cohortpairs)", "error": err.Error()})
+			c.Abort()
+			return
+		}
 
-		b := GenerateAttritionCSV(headerAndNonFilteredRow, filteredRows)
+		// concat all rows:
+		var allVariablesAttritionRows = conceptVariablesAttritionRows
```

Comment:
why not just var `allVariablesAttritionRows := append(conceptVariablesAttritionRows, customDichotomousVariablesAttritionRows...)`

Reply:
good point, let me change that.